### PR TITLE
add lint action

### LINKED ISSUE #1: action: golint
Add support for [`golint`](https://github.com/golang/lint). E.g.

```hcl
action "go-lint" {
  uses    = "sjkaliski/go-github-actions/lint@v0.1.0"
  needs   = "previous-action"
  secrets = ["GITHUB_TOKEN"]

  env {
    GO_WORKING_DIR = "./path/to/go/files"
  }
}
```

### PR DESCRIPTION
**Pull request by acburdine**
closes #1
- adds lint action

### CODE REVIEW

File: lint/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+set -e
+
+cd "${GO_WORKING_DIR:-.}"
+
+# Check if lint fails on any files.
+set +e
+OUTPUT="$(golint -set_exit_status $(find . -type f -iname '*.go'))"
```
Comment (sjkaliski) [inline lint/entrypoint.sh]: I tried testing this and the call exited with 0 and an error like:

`./path/to/file.go is in package main, not otherpackage`

I replaced the command with the following and it worked:

`OUTPUT="$(golint -set_exit_status ./...)"`

Lmk if this seems wrong!

Comment (acburdine) [reply lint/entrypoint.sh]: that makes more sense yeah. I was having a hard time thinking how I could do the `./...` approach and make it configurable, which is why I went with the same approach as gofmt. However, I think I have an idea to make it work - i'll try it and see.

File: lint/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+set -e
+
+# Exit if `golint` passes.
+if [ $SUCCESS -eq 0 ]; then
+  exit 0
+fi
+
+# Post results back as comment.
+COMMENT="#### \`golint\`
+$OUTPUT
```
[the remark this answers is not in the document]
Comment (acburdine) [reply lint/entrypoint.sh]: done!

File: lint/entrypoint.sh (hunk 1 above)
Comment (acburdine) [reply lint/entrypoint.sh]: added a `$GO_LINT_PATHS` env var which defaults to `./...` so that you can specify specific paths to lint

**Comment by sjkaliski** [comment]
@acburdine thank you! 🎉

**Pull request merged**